folly/CPortability.cpp

```diff
@@ -0,0 +1,581 @@
+
+#ifdef _MSC_VER
+#include <folly/CPortability.h>
+
+extern "C" {
+
+// Do nothing for the system log for now.
+void openlog(const char*, int, int) {}
+void closelog() {}
+void syslog(int, const char*, ...) {}
+
+void __stdcall __alarm_callback_func(HWND, UINT, UINT_PTR, DWORD) {
+  raise(SIGALRM);
+}
+
+unsigned int alarm(unsigned int seconds) {
+  SetTimer(NULL, 0, (UINT)(seconds * 1000), __alarm_callback_func);
```

Comment:
All of the portability headers were originally written to be header-only, which meant that the CPortability functions needed to only use C, however I split the bodies out when I started the second port, because otherwise I ended up having to do a recompile of most of HHVM if I changed the body of one of these functions.

As this is now able to use C++, I'll replace these with `nullptr`.